### ISSUE TITLE
Predefined Kits - Specific Assets

### ISSUE THREAD

**Issue by gilesytheking** [grokability/snipe-it#8539]
The introduction of the predefined kits will solve many problems and streamline a lot of processes for many users.

However for our use case we would also like to assign specific assets to a kit, as opposed to just the model. Some of our kits are already made up and contain a selection of individual assets.

Eg. Toolkit 1 - Asset A, Asset B, Asset C, Asset D

The Asset specific kits would allow us to checkout "toolkit 1", which in turn would checkout all the assets contained within it. An option to Check in a predefined kit of this nature would also be beneficial.

Currently we are creating a parent asset called "Toolkit 1", then checking out each asset within that kit to the parent. Our issue with this method is that the end user only receives a single checkout email for the parent asset and not for all the assets within the kit.

Alternatively we have tried checking out each asset within the kit to the user (bulk checkout or individually), however this can become time consuming with some of the larger kits.

Checking in kits / large quantities of assets is also currently time consuming, but I think bulk Check IN features deserves a separate thread.

**Comment by egudusky** [comment]
@snipe any update on this? This option would be amazing! Then the term "Predfined Kits" can actually be predefined lol. Otherwise, maybe you should rename Predfined Kits to Randomly Assigned Kits.

**Comment by snipe** [comment]
> Otherwise, maybe you should rename Predfined Kits to Randomly Assigned Kits.

For some people, the workflow is different. A graphic designer, for example, might always get a Mac Pro,  a WACOM tablet, a mouse, an external keyboard, etc. It doesn't always matter for everyone's workflow *which* of those they get.

A big part of this challenge has been meeting everyone's different expectations for what this is supposed to do.

**Comment by egudusky** [comment]
> > Otherwise, maybe you should rename Predfined Kits to Randomly Assigned Kits.
> 
> For some people, the workflow is different. A graphic designer, for example, might always get a Mac Pro, a WACOM tablet, a mouse, an external keyboard, etc. It doesn't always matter for everyone's workflow _which_ of those they get.
> 
> A big part of this challenge has been meeting everyone's different expectations for what this is supposed to do.

I understand that in some scenarios it doesn't matter which equipment is distributed, but in a lot of scenarios it does matter. The workflow you mentioned should be an option, but there should also be an option to add specific assets. How difficult would it be to add the option of specifying an Asset OR model?

**Comment by snipe** [comment]
> How difficult would it be to add the option of specifying an Asset OR model?

Pretty difficult, or else we'd have done it already ;)

**Comment by Giantbean** [comment]
TLDR; I for one think this is very relevant and vote for Kits to be expanded to handle specific assets. 

Even if a graphic designer always gets a Mac Pro, a Wacom tablet, a mouse, an external keyboard, etc. Wouldn't you want to know which Mac Pro and Wacom they received? 

I get a small asset like a mouse or keyboard serial number not mattering but as the system stands the kit seems fairly useless and making an Asset called a Designer Kit and then checking out the actual assets to that Asset makes more sense for a predefined item as opposed to randomized item kit (I agree with egudusky the current kit naming feels off based on the functionality.) It seems if the kit has a high cost item like a Mac Pro in it when I check out the kit I should have a drop down to select which one rather then randomly assigning one and If a some users workflow doesn't care which one they can pick the first one in the list or at random.
It seems the code logic would be very similar if the Kit just made a "Master Asset" and checked out the specific items to it. It could even keep its current random dump for items like keyboards and mice if people want to track those low cost items more randomly. 

In any case there is a work around of making  A Kit/Master Asset and assigning items to that asset. However, as stated gilesytheking, this limits the email and assignment of objects to the user in a way that the predefined kit does better or would if it actually assigned the right predefined asset rather then a random asset to the right user. There in lies the problem; different methodologies and needs that are hard to program in for all possible use cases and lacking clarity on limits and design of current methodologies to help users pick the method that best suits there needs yet of all possible methodologies. Having learned some of those methodologies from this post I will not be using Predefined Kits as they stand.

**Comment by stcsa-sound** [comment]
Hi all.  I've just implemented snipeit.  I think it is awesome.  I'm not sure if this is an issue with just my setup and my limited experience of installing and setting up open source software such as snipeit.  When I checkout a predefined kit it removes all of the asset names attached to that kit.  I have to re-enter them every time I check a predefined kit back in.  Is this something I have done wrong?

Thanks.

**Comment by PW-AlexG** [comment]
+1 vote up for this issue as we need to have PREDEFINED actually mean predefined. I don't know how developers thought that assigning random assets under a model type was a good idea when most users would want to know who has what asset signed out.

**Comment by snipe** [comment]
> I don't know how developers thought that assigning random assets under a model type was a good idea when most users would want to know who has what asset signed out.

In your melon heart, you thought that this was going to be constructive or additive in some way? 

“Most” is doing a lot of heavy lifting here. SOME people want “any qualifying asset”. SOME other people want a specific one. 

Making everyone happy is hard. You being a dick adds nothing to the discourse.

**Comment by PW-AlexG** [comment]
> In your melon heart, you thought that this was going to be constructive or additive in some way?
> “Most” is doing a lot of heavy lifting here. SOME people want “any qualifying asset”. SOME other people want a specific one.
> Making everyone happy is hard. You being a dick adds nothing to the discourse.

It got someone to look back at this requested issue which is listed twice under 2 different threads as a feature request. 

I actually really like Snipe and would love it if certain aspects worked a little more inclusively. Having the option to make a predefined kit be predefined by the specific asset  seems like a relatively normal request. If i need to be a bit more blunt about my thoughts on the matter to get others to continue to look at this then I will do just that. As they say, stale threads are not particularly appetizing.
